tests/Browser/Pages/Books/CreateBookTest.php

```diff
@@ -0,0 +1,242 @@
+                    ->type('name', $name)
+                    ->select('category_id', $category_id)
+                    ->type('author', $author)
+                    ->type('price', $price)
+                    ->type('employee_code', $donator_id)
+                    ->type('year', $year)
+                    ->attach('image', $image);
+
+            $this->typeInCKEditor('#cke_description iframe', $browser, $description);
+                        
+            $browser->press('Submit');
+                
+            foreach($messages as $message) {
+                $browser->assertSee($message);
+            }
+        });
+    }
+
+    /**
+     * Press button create for create success book
+     *
+     * @return void
+     */
+    public function testCreateBookSuccess()
+    {
+        $category = Category::findOrFail(1);
+        $this->browse(function (Browser $browser) use ($category) {
+            $browser->loginAs(User::find(1))
+                    ->visit('admin/books/create')
+                    ->resize(1600, 2000)
+                    ->type('name', 'Example Book')
+                    ->select('category_id', $category->name)
+                    ->type('author', 'Example Author')
+                    ->type('price', '10009')
+                    ->type('employee_code', 'AT-00001')
+                    ->type('year', '2018')
+                    ->attach('image', $this->fakeImage());
+            $this->typeInCKEditor('#cke_description iframe', $browser, 'This is a description');
+                
+            $browser->press('Submit')
+                    ->assertSee('Create Success');
+        });
+    }
+
+    /**
+     * Press button create, then create book fail
+     * 
+     * @return void
+     */
+    public function testCreateBookFail()
+    {
+        $category = Category::findOrFail(1);
+        $this->browse(function (Browser $browser) use ($category) {
+            $browser->loginAs(User::find(1))
+                    ->visit('admin/books/create')
+                    ->resize(1600, 2000)
+                    ->type('name', 'Example Book')
+                    ->select('category_id', $category->name)
+                    ->type('author', 'Example Author')
+                    ->type('price', '10009')
+                    ->type('employee_code', 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s
+                    Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lore Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lore')
+                    ->type('year', '2018')
+                    ->attach('image', $this->fakeImage());
+            $this->typeInCKEditor('#cke_description iframe', $browser, 'This is a description');
+                
+            $browser->press('Submit')
+                    ->assertSee('Create Fail. Cannot save data');
```

Comment:
tương tự test database nữa : 

`$this->assertDatabaseMissing($table, array $data);`